### PR TITLE
Firealarms now have a separate icon when alarming

### PR DESCRIPTION
**Pull request by Shadow-Quill**
## What Does This PR Do
There was no indication that a fire alarm was going off (other than the fire doors closing/sounds). I've moved the detector-disabled icon (flashing red) to be the alarming sprite, with the detector-disabled now a alternating red/green flashing.

## Why It's Good For The Game
Visual feedback from fire alarms!

## Images of changes
alarming
alarm_ex

detector disabled
detect_ex

## Testing
Went in testing server and pulled a fire alarm, as well as disabled the temperature circuit.

## Changelog
:cl:
imageadd: Added a new fire alarm sprite for when it's going off.
/:cl:

### CODE REVIEW

File: code/game/machinery/firealarm.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@ FIRE ALARM
 	desc = "<i>\"Pull this in case of emergency\"</i>. Thus, keep pulling it forever."
 	icon = 'icons/obj/monitors.dmi'
 	icon_state = "firealarm_on"
-	var/detecting = TRUE
+	var/detecting = TRUE //whether or not the fire alarm's temperature sensors are working
```
Comment (Sirryan2002) [inline code/game/machinery/firealarm.dm:14]:
```suggestion
    ///whether or not the fire alarm's temperature sensors are working
	var/detecting = TRUE 
```

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/game/machinery/firealarm.dm [hunk 2]
```diff
@@ -50,10 +51,18 @@ FIRE ALARM
...
 		icon_state = "firealarm_broken"
-	else if(stat & NOPOWER)
+		return
+	if(stat & NOPOWER)
 		icon_state = "firealarm_off"
-	else if(!detecting)
+		return
+	
+	var/area/area = get_area(src)
+	if(area.fire)
```
Comment (Sirryan2002) [inline code/game/machinery/firealarm.dm:60]:
```suggestion
	if(area?.fire)
```
oop just one thing

Comment (Shadow-Quill) [reply code/game/machinery/firealarm.dm:60]: Question marks my hated. Will do

**Review by Sirryan2002 — APPROVED** [review]

File: code/game/machinery/firealarm.dm [hunk 3]
```diff
@@ -50,10 +51,18 @@ FIRE ALARM
...
 		icon_state = "firealarm_broken"
-	else if(stat & NOPOWER)
+		return
+	if(stat & NOPOWER)
 		icon_state = "firealarm_off"
-	else if(!detecting)
+		return
+	
+	var/area/area = get_area(src)
+	if(area && area.fire)
```
Comment (hal9000PR) [inline code/game/machinery/firealarm.dm:60]:
```suggestion
	if(area.fire)
```
if the area is null this probably should runtime.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**